neqo-http3/src/recv_message.rs

```diff
@@ -337,115 +345,93 @@ impl RecvMessage {
         }
     }
 
-    fn headers_valid(&self, headers: &[Header]) -> Res<bool> {
-        let mut pseudo_header_map: HashMap<&str, bool> = HashMap::new();
-        match self.message_type {
-            MessageType::Response => {
-                pseudo_header_map.entry(":status").or_insert(false);
-            }
-            MessageType::Request => {
-                pseudo_header_map.entry(":method").or_insert(false);
-                pseudo_header_map.entry(":scheme").or_insert(false);
-                pseudo_header_map.entry(":authority").or_insert(false);
-                pseudo_header_map.entry(":path").or_insert(false);
-            }
+    fn track_pseudo(&self, name: &str, state: &mut u8) -> Res<bool> {
```

Comment:
This can be an associated function (`Self::track_pseudo` rather than `self.track_pseudo`) as it doesn't use `self`.